telemetry/definitions/commonDefinitions.json

```diff
@@ -375,6 +375,12 @@
             "type": "int",
             "description": "The metrics accepted on suggested CodeWhisperer code"
         },
+        {
+            "name": "codewhispererAutomatedTriggerState",
+            "type": "string",
+            "description": "State of cwspr auto-suggestion",
+            "allowedValues": ["activated", "deactivated"]
```

Comment:
It's noisy to create a new type for every field in each metric. Is there an existing type that can be re-used? I see `experimentState` but that is named too specifically (which is similar to the problem we want to avoid here).

There is `enabled` https://github.com/aws/aws-toolkit-common/blob/2a44db2267931adfe47af6bb37bcbd98b223884c/telemetry/definitions/commonDefinitions.json#L208
but it is a boolean. Do you plan to add more values in the future?

Reply:
Another option is to add a generic metric like `aws_modifySetting` and add a single required field `settingId` and one optional field `settingState`. Both strings.

Reply:
thanks Justin and Jaden for the great inputs, both of them work for me, will update it.

Reply:
@JadenSimon, so the caller should look something similar to `AwsTelemetry.modifySetting("cwspr_autoSuggestionActivation", "activated")`, both of the fields will be pure strings, am I correct? Thanks!

Reply:
> with the option u mentioned, so the caller should look something similar to AwsTelemetry.modifySetting(settingId = "cwspr_autoSuggestionActivation", settingState = "activated"), both of the fields will be pure strings, am I correct? Thanks!

Yup. My suggestion is intended to be more re-usable while decreasing rigidity (which is both good and bad). I'd suggest using the same "namespace" for the setting id: `codewhisperer_autoSuggestionActivation` to keep things consistent.

Eventually I'd like to have a better way to express 'subtypes' for metrics. That way we can share high-level common definitions with increasing levels of detail per feature/component without losing out on the type-safety.

Reply:
noted, yea will update changes you suggested.
Thanks Jaden!